util/fipstools/acvp/modulewrapper/modulewrapper.cc

```diff
@@ -1205,35 +1216,71 @@ static bool HashMCTSha3(const Span<const uint8_t> args[],
 
 template <const EVP_MD *(MDFunc)()>
 static bool HashMCTXof(const Span<const uint8_t> args[], ReplyCallback write_reply) {
-  // The spec for SHAKE monte carlo tests is written to be generic between a
-  // minimum and maximum output length, but the vectors obtained from ACVP
-  // allow only 1024 bits. Supporting dynamically sized MCTs would require
-  // passing the min/max output lengths to the modulewrapper, parsing ibid.,
-  // and dynamically allocating and freeing appropriately sized bufffers. To
-  // keep things simple, we defer that complexity until/if needed.
-  //
-  // https://pages.nist.gov/ACVP/draft-celi-acvp-sha3.html#name-shake-monte-carlo-test
-  const unsigned output_len = 1024/8;
-  const unsigned msg_size = 128/8;
+  const uint8_t* max_outlen_bytes = args[1].data();
+  const uint8_t* min_outlen_bytes = args[2].data();
+  const uint8_t* outlen_bytes = args[3].data();
+
+  // The various output lens are passed to modulewrapper as a length-4 byte array representing
+  // a little-endian unsigned 32-bit integer.
+  uint32_t min_output_len = 0;
+  min_output_len |= min_outlen_bytes[3] << 24;
```

Comment:
You should use `CRYPTO_load_u32_le` which will do the right thing on big/little endian computers.

Reply:
Just to be clear: `CRYPTO_load_u32_le` will always read as little endian even on big endian computers?

Reply:
Correct, it handles doing the conversion or not based on the system architecture.